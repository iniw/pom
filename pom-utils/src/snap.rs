pub use insta;

#[macro_export]
macro_rules! snap {
    ($expr:expr) => {
        $crate::snap::insta::with_settings!({prepend_module_to_snapshot => false}, {
            $crate::snap::insta::assert_debug_snapshot!($expr);
        });
    }
}
