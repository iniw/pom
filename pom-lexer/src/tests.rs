use super::*;
use pom_utils::snap;

#[test]
fn whitespace() {
    snap!(lex(" \t\r\n"));
}

#[test]
fn booleans() {
    snap!(lex("true false"));
}

#[test]
fn idents() {
    snap!(lex("w w1 _ _w1 W W1 W1_1W"));
}

#[test]
fn maximal_munch() {
    snap!(lex("truee falsee"));
}

#[test]
fn numbers() {
    snap!(lex("55 55.47 -55 +47.55"));
}

#[test]
fn operators() {
    snap!(lex("+ -"));
}

#[test]
fn errors() {
    snap!(lex("55 ^ $ 47"));
}
