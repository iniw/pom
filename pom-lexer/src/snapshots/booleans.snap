---
source: pom-lexer/src/tests.rs
expression: "lex(\"true false\")"
---
(
    [
        Token {
            kind: Bool,
            span: Span {
                start: 0,
                end: 4,
            },
        },
        Token {
            kind: Bool,
            span: Span {
                start: 5,
                end: 10,
            },
        },
        Token {
            kind: Eof,
            span: Span {
                start: 4294967295,
                end: 4294967295,
            },
        },
    ],
    Arena(
        [],
    ),
)
